crates/bevy_physics/src/d3/intersectors.rs

```diff
@@ -0,0 +1,94 @@
+use super::{geometry::*, ray::*};
+use glam::Vec3;
+
+pub struct RayHit {
+    t: f32,
+    point: Vec3,
+}
+
+impl RayHit {
+    pub fn new(t: f32, point: Vec3) -> Self {
+        Self { t, point }
+    }
+
+    pub fn t(&self) -> &f32 {
+        &self.t
+    }
+
+    pub fn point(&self) -> &Vec3 {
+        &self.point
+    }
+}
+
+pub trait RayIntersector {
+    fn intersect_ray(&self, ray: &Ray) -> Option<RayHit>;
+}
+
+impl RayIntersector for Plane {
+    fn intersect_ray(&self, ray: &Ray) -> Option<RayHit> {
+        let d = self.normal().dot(*ray.direction());
```

Comment:
Please describe what is this line doing.

Reply:
It assigns the dot product between the plane normal and the ray direction to the variable d (now changed to denominator). I don't think that needs a comment.